### PR TITLE
Campaign Entry Observer Logic

### LINKED ISSUE #253: Implement Campaign Entry Creation Observer
## Description
When an entry attached to a campaign is created, an observer should fire and create the same entries for all characters in that campaign.
This observer creates character entries on the associated characters and GM entries for the GM.
Also modify the entry controllers to pass the user's available campaigns in the entry create and edit views.

## Possible Implementation

Extend the entry observer to check if the entry belongs to a campaign, if so, then propagate the entry to the other characters in the campaign

### PR DESCRIPTION
**Pull request by IanjhPhillips**
## Description
Closes #253 
Creates a new method in the Entry Observer for creating the campaign entries
Adds logic to the created() function in the observer to handle campaign entries.

## How To Test
Open tinker
Create a campaign and attach multiple characters to it.
Create a new entry on a character via
`$someCharacter->entries()->create(['campaign_id' => $yourCampaign->id, ... , the rest of the required data])`
Observe that entries have been created on the other characters in the campaign via
`$someOtherCharacter->entries`

### CODE REVIEW

**Review by m-triassi — COMMENTED** [review]
Here's  what i can tell might be going wrong so far on this :)

File: app/Observers/EntryObserver.php [hunk 1]
```diff
@@ -73,4 +89,25 @@ public function deleting(Entry $entry)
...
+     * @param Campaign $campaign Campaign for which the entries are being created
+     * @param $entryData Array of data from the triggering entry
+     */
+    private function generateCharacterEntriesForCampaign(int $excludedCharacterId, Campaign $campaign, $entryData)
+    {
+        $characters = $campaign->characters;
+
+        foreach ($characters as $character) {
+            if ($character->id != $excludedCharacterId) {
+                $newEntry = Entry::create($entryData);
```
[the remark this answers is not in the document]
Comment (IanjhPhillips) [reply app/Observers/EntryObserver.php:105]: ah! This is exactly what I was trying to come up with, but my sleep deprived brain could not. Thank you!

**Review by m-triassi — APPROVED** [review]
Incredibly Solid! i can't find anything wrong with this, personally --  just resolve those merge conflicts? :)

**Comment by IanjhPhillips** [comment]
> Incredibly Solid! i can't find anything wrong with this, personally -- just resolve those merge conflicts? :)

resolved them!

**Review by m-triassi — APPROVED** [review]
Looks like you're still good here! Approved and ill merge now :)

**Pull request merged**